UnityProject/Assets/Scripts/Doors/DoorMasterController.cs

```diff
@@ -0,0 +1,373 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Mirror;
+using UnityEngine;
+
+//TODO: Needs unique animation for opening when powered off.
+//TODO: Need to reimplement hacking with this system. Might be a nightmare, dk yet.
+namespace Doors
+{
+	//These are used by modules when signaling to the master controller what to do when looping through modules.
+	public enum ModuleSignal
+	{
+		Continue, //Continue executing through modules.
+		Break, //Prevent any further execution, including door masters own methods.
+		SkipRemaining, //Skip the remaining modules, but continue with the door masters methods.
+		ContinueWithoutDoorStateChange, //Continue with module interactions, but the door wont change states from here on out.
+	}
+	//This is the master 'controller' for the door. It handles interactions by players and passes any interactions it need to to its components.
+	public class DoorMasterController : NetworkBehaviour, IPredictedCheckedInteractable<HandApply>
```

Comment:
since we're doing nothing with prediction we maybe should change the interface for ``ICheckedInteractable``

Reply:
Sounds good to me.